### PR TITLE
Add missing emitc conversions for vm/test/arithmetic_ops.mlir

### PR DESCRIPTION
**Pull request by simon-camp**

### CODE REVIEW

**Comment by simon-camp** [comment]
The remaining ops missing are `vm.cond_br` and `vm.fail`. These will go into the [printer](https://github.com/google/iree/blob/969e8e89138b630c2e98f8bedfa7609243a38fdf/iree/compiler/Dialect/VM/Target/C/CModuleTarget.cpp) for the time being.

**Review by marbre — APPROVED** [review]
LGTM, but I would like to ask for another opinion and we need an approval from a Googler anyway :)

File: iree/compiler/Dialect/VM/Target/C/CModuleTarget.cpp
Comment (GMNGeoffrey) [inline iree/compiler/Dialect/VM/Target/C/CModuleTarget.cpp:253]: For nested passes, I think you want runPipeline here, using the newish pipeline-in-a-pass infra. We have an example intree at https://github.com/google/iree/blob/main/iree/compiler/Dialect/HAL/Transforms/TranslateExecutables.cpp

File: iree/vm/c_funcs.h
Comment (GMNGeoffrey) [inline iree/vm/c_funcs.h:27]: I don't really get why emitting this function would ever be useful.

File: iree/compiler/Dialect/VM/Conversion/VMToEmitC/ConvertVMToEmitC.cpp
Comment (simon-camp) [reply iree/compiler/Dialect/VM/Conversion/VMToEmitC/ConvertVMToEmitC.cpp:30]: Yeah, it's the absence of attributes that matters though, so I'll go with `NoAttributeOpConversion`.

File: iree/compiler/Dialect/VM/Target/C/CModuleTarget.cpp
Comment (simon-camp) [reply iree/compiler/Dialect/VM/Target/C/CModuleTarget.cpp:253]: This then also affects the bytecode module target. So I'd suggest to change it in a follow up PR.

File: iree/vm/c_funcs.h
Comment (simon-camp) [reply iree/vm/c_funcs.h:27]: I see multiple ways of printing constant operations (not having thought about ref types).

1. Convert to calls to "pseudo" functions like it is now
2. Handle this in the printer (CModuleTarget)
3. Convert to std.const operations and let emitc do the printing
4. Like 3 but define an emitc.const op

Comment (marbre) [reply iree/vm/c_funcs.h:27]: Yeah, I agree that the way this handled now isn't very convenient. However, I would like to avoid to bloat the printer (option 2) and would prefer to handle the const op more MLIR-ish. For MLIR-HLO, I have chosen option 3, see the forked [MHLOToStd](https://github.com/iml130/mlir-emitc/tree/main/lib/Conversion) conversion. So far I don't see strong pros of an emitc.const op (option 4). A conversion might be more straight forward. I would suggest to handle this in a follow up PR.

**Review by marbre — APPROVED** [review]

**Review by GMNGeoffrey — APPROVED** [review]

**Pull request merged**
